core/src/main/java/tc/oc/pgm/util/xml/parsers/FilterBuilder.java

```diff
@@ -27,6 +29,26 @@ public FilterBuilder dynamic(Class<? extends Filterable<?>> type) {
     return this;
   }
 
+  public FilterBuilder respondsTo(@NotNull Class<? extends Filterable<?>> type) {
+    validate((filter, node) -> filters.validate(
+        filter,
+        (definition, n) -> {
+          if (!definition.respondsTo(type)) {
+            throw new InvalidXMLException(
+                "Expected a filter that can respond to "
+                    + type.getSimpleName()
+                    + ". The filter "
+                    + definition.getDefinitionType().getSimpleName()
+                    + " only responds to "
+                    + Filterables.scope(definition).getSimpleName()
```

Comment:
This can and will throw for filters that do not belong to any scope, which is possible in non-dynamic filters.